### PR TITLE
Add a side-efect component for ScrollPosition sync

### PR DESCRIPTION
**Pull request by schwers**
👓 @nramadas

### CODE REVIEW

File: src/app/side-effect-components/ScrollPositionSync.js [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    super(props);
+
+    const { url, scrollY } = props;
+    this.state = { url, scrollY };
+    this.callMeasureScrollPosition = this.callMeasureScrollPosition.bind(this);
+    this.measureScrollPosition = this.measureScrollPosition.bind(this);
+  }
+
+  componentDidMount() {
+    window.addEventListener('scroll', this.measureScrollPosition);
```
Comment (nramadas) [inline src/app/side-effect-components/ScrollPositionSync.js]: should this be `this.callMeasureScrollPosition`?

File: src/app/side-effect-components/ScrollPositionSync.js [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+  (url, scrollY) => ({ url, scrollY }),
+);
+
+const mapDispatchToProps = dispatch => ({
+  onSaveScrollPosition(url, scrollY) {
+    dispatch(scrollPositionActions.save(url, scrollY));
+  },
+});
+
+export const ScrollPositionSync = connect(
```
Comment (nramadas) [inline src/app/side-effect-components/ScrollPositionSync.js]: :no_bell: out of curiosity, why not `export default`?

File: src/app/side-effect-components/ScrollPositionSync.js [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+}
+
+// We want to ignore urls that are for overlay menus, those render above content
+// so they shouldn't have their own scroll position state.
+const scrollUrl = page => {
+  const { url, hashParams, queryParams } = page;
+  return urlFromPage({
+    url,
+    hashParams,
+    queryParams: omit(queryParams, [OVERLAY_MENU_PARAMETER ]),
```
Comment (nramadas) [inline src/app/side-effect-components/ScrollPositionSync.js]: :no_bell: extra space after `OVERLAY_MENU_PARAMETER`

**Comment by nramadas** [comment]
Patch is good, just one thing that might be a bug, otherwise looks good.
:nail_care:

File: src/app/side-effect-components/ScrollPositionSync.js (hunk 1 above)
Comment (schwers) [inline src/app/side-effect-components/ScrollPositionSync.js]: Yes, thanks for catching my refactoring bug

**Comment by schwers** [comment]
💇

**Comment by nramadas** [comment]
:fish:

**Pull request merged**